### PR TITLE
Added AD FS auth field details

### PR DESCRIPTION
**Pull request by rashminandasahoo**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 1]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+  
 
-Click **Continue**.
+```mdx-code-block
+</TabItem>
+<TabItem value="adfs" label="ADFS Client Credentials with Certificate">
+```
+  To use AD FS credentials for authentication, do the following:
+  
+  1. Select an existing secret or create one that has the resource id of ServiceNow as its value for the **Resource ID** field.
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:74]: ```that has the resource identifier of ServiceNow configured in ADFS```

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 2]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+  2. Select an existing secret or create one that has the application (client) id assigned to your application by AD FS for the **Client ID** field.
+   
+```mdx-code-block
+<img src={adfs_clientid} alt="adfs-clientid" height="150" width="400"/>
+```
 
-Select the Harness Delegate(s) to use when making a connection to ServiceNow using this Connector.
+
+  3. Select an existing encrypted file or create one that has the certificate for the **Certificate** field. Certificates are digital documents that contain a public key and other information about the certificate holder.
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:82]:
```
This public x509 certificate is the one which is used for signing JWT tokens by your application
```

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 3]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+  4. In **Private Key**, create a new secret or choose an existing one that has the AD FS private key as the value.
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:83]: ```This key is the private RSA key corresponding to certificate uploaded in **Certificate **field ```

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 4]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+  5. In **ADFS URL**, enter the AD FS URL.
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:84]: enter base ADFS url

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 5]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:85]: Can these additional notes be added 
1. Details of Client Credentials with certificate can be found here [AD FS OpenID Connect/OAuth flows and Application Scenarios](https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/overview/ad-fs-openid-connect-oauth-flows-scenarios#second-case-access-token-request-with-a-certificate-1).
2. We only support RSA encoded (Pkcs8 format) private keys . 
3. We only support x509 certificates

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 6]
```diff
@@ -7,55 +7,102 @@ helpdocs_category_id: o1zhrfo8n5
...
+<img src={adfs_clientid} alt="adfs-clientid" height="150" width="400"/>
+```
 
-Select the Harness Delegate(s) to use when making a connection to ServiceNow using this Connector.
+
+  3. Select an existing encrypted file or create one that has the certificate for the **Certificate** field. Certificates are digital documents that contain a public key and other information about the certificate holder.
+  4. In **Private Key**, create a new secret or choose an existing one that has the AD FS private key as the value.
+  5. In **ADFS URL**, enter the AD FS URL.
+  
+```mdx-code-block
```
Comment (naman-goenka) [inline docs/platform/7_Connectors/connect-to-service-now.md:86]: Please add that ADFS support is behind this CDS_SERVICENOW_ADFS_AUTH FF

**Review by naman-goenka — DISMISSED** [review]

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 7]
```diff
@@ -7,55 +7,113 @@ helpdocs_category_id: o1zhrfo8n5
...
-Enter your credentials. For **API Key**, use a Harness [Text Secret](../6_Security/2-add-use-text-secrets.md). 
+```mdx-code-block
+</TabItem>
+<TabItem value="adfs" label="ADFS Client Credentials with Certificate">
+```
+  
 
-Click **Continue**.
+:::note
+  Currently, this feature is behind the feature flag `CDS_SERVICENOW_ADFS_AUTH`. Contact Harness Support to enable the feature.
```
Comment (aimurphy) [inline docs/platform/7_Connectors/connect-to-service-now.md:79]:
```suggestion
Currently, this feature is behind the feature flag `CDS_SERVICENOW_ADFS_AUTH`. Contact Harness Support to enable the feature.
```

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 8]
```diff
@@ -7,55 +7,113 @@ helpdocs_category_id: o1zhrfo8n5
...
+  2. Select an existing secret or create one that has the application (client) id assigned to your application by AD FS for the **Client ID** field.
+   
+```mdx-code-block
+<img src={adfs_clientid} alt="adfs-clientid" height="150" width="400"/>
+```
+
+
+  3. Select an existing encrypted file or create one that has the certificate for the **Certificate** field. This is a `X509` format certificate used for signing JWT tokens by your application.
+  4. In **Private Key**, create a new secret or choose an existing one that has the AD FS private key as the value. This key is the private RSA key corresponding to certificate uploaded in the **Certificate **field.
+  5. In **ADFS URL**, enter the base AD FS URL.
```
Comment (aimurphy) [inline docs/platform/7_Connectors/connect-to-service-now.md:95]:
~~~suggestion
To use AD FS credentials for authentication, do the following:
  
1. Select an existing secret or create one that has the resource identifier of ServiceNow configured in ADFS as its value for the **Resource ID** field.
2. Select an existing secret or create one that has the application (client) id assigned to your application by AD FS for the **Client ID** field.
   
```mdx-code-block
<img src={adfs_clientid} alt="adfs-clientid" height="150" width="400"/>
```

3. Select an existing encrypted file or create one that has the certificate for the **Certificate** field. This is a `X509` format certificate used for signing JWT tokens by your application.
4. In **Private Key**, create a new secret or choose an existing one that has the AD FS private key as the value. This key is the private RSA key corresponding to certificate uploaded in the **Certificate **field.
5. In **ADFS URL**, enter the base AD FS URL.
~~~

File: docs/platform/7_Connectors/connect-to-service-now.md [hunk 9]
```diff
@@ -7,55 +7,113 @@ helpdocs_category_id: o1zhrfo8n5
...
 
 Click **Save and Continue**.
 
 Harness tests the connection.
 
 ![](./static/connect-to-service-now-44.png)
+
 Click **Finish**.
 
 The ServiceNow Connector is listed in Connectors.
```
Comment (aimurphy) [inline docs/platform/7_Connectors/connect-to-service-now.md:119]:
```suggestion
The ServiceNow connector is listed in Connectors.
```

**Review by aimurphy — DISMISSED** [review]
approved w a few small comments

**Pull request merged**